What's New v101 (#1895)
MicrosoftDocs/edge-developer: microsoft-edge/devtools-guide-chromium/whats-new/2022/04/devtools-101.md

If you're on Windows, Linux, or macOS, consider using the Canary preview channel of Microsoft Edge as your default development browser.  The preview channels of Microsoft Edge (Beta, Dev, and Canary) give you access to the latest features of Microsoft Edge DevTools.

To report a problem with DevTools or ask for a new feature, check out the new [MicrosoftEdge > DevTools](https://github.com/MicrosoftEdge/DevTools) repo.

<!-- ====================================================================== -->
## The Console tool is now available in Visual Studio Code

<!-- Title: Visual Studio Code extension now with Console tool -->
<!-- Subtitle: See log messages, run JavaScript, and use Console APIs right next to your code in Visual Studio Code. -->

> Using the Console is one of the biggest use-cases of DevTools.  That's why we made it easier for you to access the Console tool by turning it on in the Microsoft Edge Developer Tools extension for Visual Studio Code.  This allows you to see your log messages right next to your code.  You can also try out some JavaScript,<!--enter JavaScript code,--> access the window object, and use the Console API to access the DOM.  You can use the Console as the main tool, or use it together with the **Elements** and **Network** tools.

Author: @codepo8 @zoherghadyali 
Alt wording:
try out some JavaScript, 
enter JavaScript code, 
try entering JavaScript expressions, 
test lines of JavaScript code,

> Using the Console is one of the biggest use-cases of DevTools.  That's why we made it easier for you to access the Console tool by turning it on in the Microsoft Edge Developer Tools extension for Visual Studio Code.  This allows you to see your log messages right next to your code.  You can also try out some JavaScript,<!--enter JavaScript code,--> access the window object, and use the Console API to access the DOM.  You can use the Console as the main tool, or use it together with the **Elements** and **Network** tools.

Reviewer: ```suggestion
The Console is one of the most popular tools in DevTools.  You can now access the Console tool in the Microsoft Edge Developer Tools extension for Visual Studio Code.  This allows you to see your log messages right next to your code.  You can also try out some JavaScript,<!--enter JavaScript code,--> access the window object, use the [Console API](https://docs.microsoft.com/microsoft-edge/devtools-guide-chromium/console/api), and [Console utilities](https://docs.microsoft.com/microsoft-edge/devtools-guide-chromium/console/utilities) to access the DOM.  You can use the Console as the main tool, or use it together with the **Elements** and **Network** tools.
```

I know we don't like in-line links (I think) but I also think this is a valuable place to link to the Console API reference we have documented here: https://docs.microsoft.com/microsoft-edge/devtools-guide-chromium/console/api and the Console utilities we have documented here: https://docs.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/console/utilities. The rest of my changes are more about not saying "Look what we did for you" and instead just saying "You are now able to..."

The Console tool in the Drawer of the Edge DevTools extension for Visual Studio Code:

![The Console tool in the Drawer of the Edge DevTools extension for Visual Studio Code.](devtools-101-images/console-tool-vsc-2.png)

For more information, see [Console integration](../../../../visual-studio-code/microsoft-edge-devtools-extension.md#console-integration) in _Microsoft Edge DevTools extension for Visual Studio Code_.

<!-- ====================================================================== -->
## Source maps are now cached with IndexedDB

<!-- Title: Speed up debugging with cached sourcemaps -->
<!-- Subtitle: DevTools now caches sourcemaps with IndexedDB, reducing the need to fetch sourcemaps on refresh. -->

In previous versions of Microsoft Edge, if you were hosting large sourcemap files on your server or in [Azure Artifacts symbol server](../../../javascript/publish-source-maps-to-azure.md), DevTools would fetch these sourcemaps every time the page refreshed.  Fetching these sourcemaps repeatedly created additional network requests for DevTools and slowed down debugging time.

> In Microsoft Edge 101, DevTools caches sourcemaps by using IndexedDB.  Upon refreshing the page, sourcemaps are now served from the IndexedDB cache instead of being requested again from your server or Azure Artifacts symbol server.  Use the **Load status** column in the [Source Maps Monitor tool](../../../source-maps-monitor/source-maps-monitor-tool.md) to verify that sourcemaps are being loaded from the cache.  To clear the IndexedDB cache and force DevTools to reload sourcemaps, run the **Clear source maps cache** command in the [Command Menu](../../../command-menu/index.md).

Reviewer: ```suggestion
In Microsoft Edge 101, DevTools caches sourcemaps by using [IndexedDB](https://developer.mozilla.org/docs/Web/API/IndexedDB_API).  Upon refreshing the page, sourcemaps are now served from the IndexedDB cache instead of being requested again from your server or Azure Artifacts symbol server.  Use the **Load status** column in the [Source Maps Monitor tool](../../../source-maps-monitor/source-maps-monitor-tool.md) to verify that sourcemaps are being loaded from the cache.  To clear the IndexedDB cache and force DevTools to reload sourcemaps, run the **Clear source maps cache** command in the [Command Menu](../../../command-menu/index.md).
```

Minor tweak.  Maybe some readers don't know what IndexedDB is and why it's useful so we could link to MDN optionally

In addition, a new setting has been added to the Network tool: **Include pending requests in HAR file**.  If you need to record **Pending** network requests when exporting a HAR file in the Network tool, turn on the **Include pending requests in HAR file** setting.  The **Status text** column will now include the **(pending)** state for any pending network requests.

![The "Status text" column in the Network tool.](devtools-101-images/status-text-column-network-tool.png)

To learn more about the available columns in the request log in the **Network** tool, see [Add or remove columns](../../../network/reference.md#add-or-remove-columns) in _Network features reference_.

<!-- ====================================================================== -->
## Updated UI and camera controls in 3D View

<!-- Title: Improvements to the 3D View tool -->
<!-- Subtitle: Check out 3D View for updates to the UI and smoother camera controls. -->

> You asked and we answered!  For ease of use, there are now two buttons, **Pan view** (![The 'Pan view' icon.](devtools-101-images/pan-view-icon.png)) and **Rotate view** (![The 'Rotate view' icon.](devtools-101-images/rotate-view-icon.png)), that you can use to switch between panning around the canvas or rotating the canvas.  Just select a mode, and you can drag in the desired direction with your mouse or arrow keys.

Reviewer: ```suggestion
In Microsoft Edge 101, there are now two buttons, **Pan view** (![The 'Pan view' icon.](devtools-101-images/pan-view-icon.png)) and **Rotate view** (![The 'Rotate view' icon.](devtools-101-images/rotate-view-icon.png)), in the **3D View** tool that you can use to switch between panning around the canvas or rotating the canvas.  Just select a mode, and you can drag in the desired direction with your mouse or arrow keys.
```

> We also cleaned up the UI for a more uniform and intuitive experience by organizing the panes more clearly and fixing accessibility issues.

Reviewer: ```suggestion
The UI now provides a more uniform and intuitive experience by organizing the panes more clearly and fixing accessibility issues.
```

<!-- ====================================================================== -->
## The Console tool is now available in Visual Studio Code

<!-- Title: Visual Studio Code extension now with Console tool -->
<!-- Subtitle: See log messages, run JavaScript, and use Console APIs right next to your code in Visual Studio Code. -->

The Console is one of the most popular tools in DevTools.  You can now access the Console tool in the Microsoft Edge Developer Tools extension for Visual Studio Code.  This integrated DevTools Console allows you to see your log messages right next to your code.  Using the Console, you can also:
*  Try out some JavaScript statements.
*  Access the window object.
*  Call the methods of the `Console` object, as described in [Console object API Reference](../../../console/api.md).
*  Use the Console utilities to access the DOM, as described in [Console tool utility functions and selectors](../../../console/utilities.md).

You can use the Console as the main tool, or use it together with the **Elements** and **Network** tools.

The Console tool in the main toolbar of the Microsoft Edge DevTools extension for Visual Studio Code:

> ![The Console tool in the main toolbar of the Microsoft Edge DevTools extension for Visual Studio Code.](devtools-101-images/console-tool-vsc-1.png)

Author: nice todo: revert my VS Code to default color theme, retake screenshots
@codepo8 - i might ask for your todo app

To learn more about the available columns in the request log, see [Network features reference > Add or remove columns](../../../network/reference.md#add-or-remove-columns).

<!-- ====================================================================== -->
## Updated UI and camera controls in 3D View

<!-- Title: Improvements to the 3D View tool -->
<!-- Subtitle: Check out 3D View for updates to the UI and smoother camera controls. -->

In Microsoft Edge 101, the **3D View** tool has two new buttons: **Pan view** (![The 'Pan view' icon.](devtools-101-images/pan-view-icon.png)) and **Rotate view** (![The 'Rotate view' icon.](devtools-101-images/rotate-view-icon.png)).  Use these buttons to switch between panning around the canvas or rotating the canvas.  Click the **Pan view** or **Rotate view** button, and then drag in the desired direction with your mouse, or use the arrow keys.

The UI now provides a more uniform and intuitive experience.  The panes are organized more clearly, and accessibility issues have been fixed.

![Updated UI and camera controls in 3D View.](devtools-101-images/3d-view-ui-camera-ctrls.png)

> See [Navigate z-index, DOM, and layers using the 3D View tool](../../../3d-view/index.md).

Reviewer 2: Will this document be updated by the time this comes out?

Author: updated 3D View article.
